### DISCUSSION TITLE
Where is the View tree created?

### DISCUSSION THREAD

**Discussion by test3211234** [lapce/floem#610] [Q&A]
I tried to search for id.add_child but there's basically no occurrences. Also, I really think this library should have an explanation or diagram of its architecture. The code base is very confusing. Is it based on the old Xilem mostly? What parts did you take from different libraries? I noticed there's a ChangeFlags but I don't see it being used in the return of View functions. I also saw you got rid of the Widget trait, but it didn't even seem to do the same things as in Xilem. There must be a lot of inspirations because making a library this full-fledged is really hard. Great job, by the way. There's also a lot of thread local RefCells, not sure how bad that affects performance or if its needed.

**Comment by jrmoulton** [comment]
The docs are a work in progress. 

views are created in the function that builds them. The `id.set_children` function is more common that `add_child`. 
https://github.com/lapce/floem/blob/4510c4bf5f615b8a4c150a16c267071318958fc2/src/views/container.rs#L17C5-L17C35
